torch/csrc/utils/tensor_types.cpp

```diff
@@ -20,6 +20,9 @@ static const char* backend_to_string(const at::Type& type) {
     case at::Backend::CUDA: return "torch.cuda";
     case at::Backend::SparseCPU: return "torch.sparse";
     case at::Backend::SparseCUDA: return "torch.cuda.sparse";
+    // We split complex into its own backend, but keeping it the same here for now
+    case at::Backend::ComplexCPU: return "torch";
+    case at::Backend::ComplexCUDA: return "torch.cuda";
```

Comment:
There's only one use site of this function, in `type_to_string`. I see you have adjusted it in #19295 but the adjustment looks a bit questionable. Wouldn't it make more sense to stop having a "type to string" operation, and instead, just pass the trinity (device type, layout, dtype) and render that way?

Reply:
Basically, after your change here, this function doesn't make "semantic" sense anymore. So I'd much prefer if we got rid of this function entirely. Can be a PR on top, if necessary.